### ISSUE TITLE
Make sure storage service auto-version-increment after a release is working as expected - Update package version is failing

### ISSUE THREAD

**Issue by ahsonkhan** [Azure/azure-sdk-for-cpp#1404]
Currently, the four storage releases seem to be triggered and running in parallel and causing the pushes to the auto-version-increment PR to fail. We should make sure the mechanism is either run sequentially or resilient to being run in parallel where the branch might be behind in commits compare to remote.

This may require changes to the storage ci.yml to enable.

See https://github.com/Azure/azure-sdk-for-cpp/pull/1373#discussion_r557224887 for context.

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=686888&view=logs&j=eef82e6a-4819-5628-5f9f-b00d0ac68cdd&t=7eb8787a-cc9a-5cf2-7b65-b14ff29415e3&l=122
```text
git checkout -b increment-package-version-storage-686888
Switched to a new branch 'increment-package-version-storage-686888'
git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" commit  -am "Increment package version after release of azure-storage-blobs"
[increment-package-version-storage-686888 96b7fa9] Increment package version after release of azure-storage-blobs
 2 files changed, 47 insertions(+), 44 deletions(-)
git push azure-sdk-fork increment-package-version-storage-686888 
To https://github.com/azure-sdk/azure-sdk-for-cpp.git
 ! [rejected]        increment-package-version-storage-686888 -> increment-package-version-storage-686888 (non-fast-forward)
error: failed to push some refs to 'https://github.com/azure-sdk/azure-sdk-for-cpp.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
Git push failed with LASTEXITCODE=1 Need to fetch and rebase: attempt number=1
git fetch azure-sdk-fork
git diff increment-package-version-storage-686888~ increment-package-version-storage-686888 --output C:\Users\VssAdministrator\AppData\Local\Temp\tmp22A7.tmp
git reset --hard azure-sdk-fork/increment-package-version-storage-686888
HEAD is now at af2fbc3 Increment package version after release of azure-storage-common
git apply -C0 C:\Users\VssAdministrator\AppData\Local\Temp\tmp22A7.tmp
C:/Users/VssAdministrator/AppData/Local/Temp/tmp22A7.tmp:68: trailing whitespace.
## 1.0.0-beta.1 
Context reduced to (1/1) to apply fragment at 5
Context reduced to (0/0) to apply fragment at 97
error: patch failed: sdk/storage/azure-storage-blobs/CHANGELOG.md:126
error: sdk/storage/azure-storage-blobs/CHANGELOG.md: patch does not apply
D:\a\1\s\eng\common\scripts\git-branch-push.ps1 : Unable to apply diff file LASTEXITCODE=1, see command output above.

```

cc @Jinming-Hu, @katmsft, @danieljurek, @weshaggard

**Comment by weshaggard** [comment]
@chidozieononiwu please work with @danieljurek  to see what went wrong here. There must be some common files being changed which is causing a merge conflict between the different storage packages.

**Comment by danieljurek** [comment]
This was an interesting failure mode for `eng/common`. 

The issue that presented: In services releasing multiple packages changes to the CHANGELOG.md file always went to the package in the service directory whose name came first (i.e. `azure-storage-blobs`) instead of the expected package folder. 

Cause: This is because `Language-Settings.ps1` had an improper implementation of `Get-cpp-PackageInfoFromRepo`. However, it was never intended that an implementation of `Get-cpp-PackageInfoFromRepo` was to be written at the time that I did implement it. 

The sequence of events went: 

1. The `Update-Change-Log.ps1` file was removed from `eng/common` ([delete in the C++ repo](https://github.com/Azure/azure-sdk-for-cpp/commit/f2d3befda1443477101ec42d81c462dfe22063e5))
2. I was working on the PR to enable vcpkg and noticed that updating changelog was broken, couldn't find `Update-Change-Log.ps1` and tried to implement `Get-cpp-PackageInfoFromRepo` by inferring its functionality from the signature. ([initial pass at fix in my C++ fork](https://github.com/danieljurek/azure-sdk-for-cpp/commit/b16968b837379981eb3f5a84a7d8729eaad18cce))
3. The next day, `Update-Change-Log.ps1` was reintroduced to `eng/common` ([in the C++ repo](https://github.com/Azure/azure-sdk-for-cpp/commit/ae4b70875f3915c67f0166b4d3754adf52bac5ef))

Had the timing of these changes been slightly different we wouldn't have encountered this exact problem. 

Two things went wrong: 

* We should've used `Update-Change-Log.ps1` but it was deleted then re-added the next day. Possible fix: When deleting files search all affected codebases for usage of those files (easier said than done) 
* Shouldn't have implemented `Get-cpp-PackageInfoFromRepo` without first knowing more about it. Possible fix: When encountering a blocking change in `eng/common` it makes more sense to huddle with the team to ensure the correct functionality is implemented or the blocking change reverted. 

cc @weshaggard
